synapse/lib/stormtypes.py

```diff
@@ -3654,50 +3661,69 @@ def getObjLocals(self):
             'gen': self._methQueueGen,
             'del': self._methQueueDel,
             'get': self._methQueueGet,
+            'byname': self._methQueueGetByName,
             'list': self._methQueueList,
         }
 
-    async def _methQueueAdd(self, name):
+    async def _methQueueAdd(self, name, iden=None):
+        name = await tostr(name)
+        iden = await tostr(iden, noneok=True)
 
-        info = {
-            'time': s_common.now(),
+        qdef = {
             'creator': self.runt.user.iden,
+            'iden': iden,
+            'name': name,
         }
 
-        todo = s_common.todo('addCoreQueue', name, info)
-        gatekeys = ((self.runt.user.iden, ('queue', 'add'), None),)
-        info = await self.dyncall('cortex', todo, gatekeys=gatekeys)
+        if iden is not None:
+            if not s_common.isguid(iden):
+                raise s_exc.BadArg(name='iden', arg=iden, mesg=f'Argument {iden} it not a valid iden.')
 
-        return Queue(self.runt, name, info)
+        self.runt.confirm(('queue', 'add'))
+        info = await self.runt.view.core.addCoreQueue(qdef)
+        iden = info.get('iden')
+        return Queue(self.runt, name, iden, info)
 
     @stormfunc(readonly=True)
-    async def _methQueueGet(self, name):
-        todo = s_common.todo('getCoreQueue', name)
-        gatekeys = ((self.runt.user.iden, ('queue', 'get'), f'queue:{name}'),)
-        info = await self.dyncall('cortex', todo, gatekeys=gatekeys)
+    async def _methQueueGet(self, iden):
+        iden = await tostr(iden, noneok=True)
```

Comment:
noneok is only for optional parameters.
```suggestion
        iden = await tostr(iden)
```